homeassistant/components/sensor/bart.py

```diff
@@ -0,0 +1,256 @@
+"""
+Support for the BART (Bay Area Rapid Transit) API.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.bart/
+"""
+import logging
+from datetime import timedelta
+
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+ICON = 'mdi:train'
+
+# Return cached results if last scan was less then this time ago.
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
+
+
+def parse_api_response(raw_xml):
+    """Parse the BART API response."""
+    import xml.etree.ElementTree as etree
+    if isinstance(raw_xml, bytes):
+        parsed_xml = etree.fromstring(
+            raw_xml, parser=etree.XMLParser(
+                encoding='utf-8'))
+    else:
+        parsed_xml = etree.parse(raw_xml)
+    return parsed_xml
+
+
+def get_bart_xml(url):
+    """Helper function to query the BART API."""
+    from urllib.request import urlopen
```

Comment:
Please use the `requests` library. Also, you only have to put imports in methods if they are for 3rd party dependencies. Standard lib imports have to be at the top of the file.